### ISSUE TITLE
Readcomiconline.to/KissManga 403 & 503 errors 

### ISSUE THREAD

**Issue by HereticWilson** [inorichi/tachiyomi-extensions#258]
App version:
R1374, readcomic extension version 1.2.1
Issue/Request:
Source not loading
Steps to reproduce (if applicable)

Other details:
Tried uninstalling and reinstalling the extension, did not work. Getting the same errors across multiple devices with the same app and extension versions.

**Comment by CarlosEsco** [comment]
you need to open the website in the browser.  then try in app.  The source forces captchas at times.

**Comment by mioaki** [comment]
Which browser do i need to open the website in? I tried with chrome and still doesn't work. The 503 error comes right when we chose "latest" or "browse" at the catalogues selection.

**Comment by CarlosEsco** [comment]
On my test device i clicked browse, got the 503.  Then opened chrome went to readcomiconline.to.  Once it loaded switched back to app and latest/browse worked fine.

**Comment by mioaki** [comment]
Yes. It works after the third time i open chrome and go to the website. No idea why the first and second time didn't work.

**Comment by HereticWilson** [comment]
@CarlosEsco Do you mean the thing where it says "wait for 5 seconds"? Because I've never noticed a captcha on the site. I've also been using the extension for around a couple weeks and this problem only recently started happening, while that "wait 5 seconds" screen has been around for months (at least).

**Comment by CarlosEsco** [comment]
For some reason the test device brought a google captcha that's why it got 503.  In browser going past the normal 5 secs it's worked fine after switching back

**Comment by HereticWilson** [comment]
Sorry, I'm not sure what that means. Does that mean my problem is being caused by something different? I tried visiting in Chrome and then opening the app back up, still got the 403 error.

**Comment by CarlosEsco** [comment]
So where are you getting 403?

**Comment by HereticWilson** [comment]
Both in my library, trying to read a bookmarked title, and just trying to browse the catalogue.

**Comment by CarlosEsco** [comment]
On the library ones. If you go to info page and 3 dot menu, and open in browser does it load?  After it loads if you go back can you read in app?

**Comment by CarlosEsco** [comment]
@HereticWilson there is something up with cloudflare.  Here is a fix.  

Go to catalog browse readcomiconline.  When it fails click retry it should load library now.  Now your library comics will work

**Comment by metroxx** [comment]
How similar problem. Mine only solved when i read a chapter on the website (mine was with kissmanga, but they use cloudflare too so probably this is a problem on any of the same type of site)

**Comment by CarlosEsco** [comment]
So those two are the only cloudflare with issue but they both have the anti bot.  Other cloudflare sites work fine

**Comment by HereticWilson** [comment]
@CarlosEsco I hate to say it, but neither of those fixes is working on my phone. I go to browse the catalogue, get the 403 error. I hit retry on the popup, it tries to load, then gives me the 503 error. As for loading the title in browser, that part works, I can browse the comic in Chrome, but when I try to go back to the title in the app it still won't load.

**Comment by drewbizc** [comment]
I also have this issue, but after messing around i was able to what seems like momentarily solve it. I loaded the site on chrome opened up a random comic, back to app. Went to browse let it get the first error, hit the upside down pyramid in the top right(search filters), and hit reset. Hit retry and got it working. So far it hasn't broken on me yet. (Tried on and off WIFI)

**Comment by HereticWilson** [comment]
@drewbizc God this is annoying, I still can't get this to work. Makes it even worse that it was working for weeks for me with the F-droid dev version.

**Comment by LakotaGray** [comment]
It has nothing to do with app being from fdroid or GitHub. It’s the source itself. Specifically cloudflare. If it’s still not allowing you to enter after having loaded through phone browser, do you use any ad-blockers? If so turn them off. Do you use a VPN? Try turning it off. Try the suggested solution on both your cell network and then on WiFi.

**Comment by metroxx** [comment]
Settings>advanced
Try all option. Friend of mine had similar problem. For him this was the solution

**Comment by CarlosEsco** [comment]
To add to above. On discord this is what was working for some. Clear cookies in app then go to a chapter in phone browser. Then go back to app and try the source

**Comment by HereticWilson** [comment]
Oh damn that actually works!

**Comment by Korpip** [comment]
The workaround is not working for me currently.  This is on a fresh install of the application on a clean install of Android 8.1.0 using Chrome to browse the site.  This extension is using the default header right?  Does this application use its own user agent?

**Comment by drewbizc** [comment]
Similar to @Korpip i am also having the 503/403 issues even after using all workarounds listed here

1. Using Settings>Advanced>All Clear Options
2. Going to the site directly and opening up a chapter
3. Uninstalling, Reinstalling the extension
4. My Own Workaround Of "Resetting The Search"

All these have failed both on and off WIFI.

**Comment by GyaragaX** [comment]
Echoing @drewbizc and @Korpip I've been trying all day to get the workaround to function. No dice. I've tried clearing cookies, loading up Kissmanga.com and Readcomiconline.to, my preferred sources, in a browser. I use Firefox, but I've also tried in Chrome, in case that helped. It didn't. I've uninstalled and reinstalled the extensions. I've disabled ad blocking (though running it previously was not an issue).
